typing/typemod.ml

```diff
@@ -2592,7 +2616,9 @@ and type_module_aux ~alias sttn funct_body anchor env smod =
       Shape.abs funct_shape_param body_shape
   | Pmod_apply _ | Pmod_apply_unit _ ->
       type_application smod.pmod_loc sttn funct_body env smod
-  | Pmod_constraint(sarg, smty) ->
+  | Pmod_constraint(sarg, smty, smode) ->
+      check_no_modal_modules ~env smode;
+      let smty = Option.get smty in
```

Comment:
We need a check in `ast_invariants.ml` to make sure that either `smty` is present or `smode` is non-empty. Otherwise it's possible to hit this exception case.